File: prototypeLeger/app/src/main/java/com/example/scrabbleprototype/model/UserPreferences.kt
package com.example.scrabbleprototype.model

import com.example.scrabbleprototype.R

enum class Language {
    French,
    English,;

    override fun toString(): String {
        if(super.ordinal == 0) return "Français"
        return super.toString()
    }
}

class UserPreferences {
    var appThemeSelected: String = "Default"

    val boardItems = arrayListOf(Item("Par défaut", 0, R.color.light_green, "Description du thème par défaut : vert clair et turquoise"))
    var boardItemSelected: Item = boardItems.first()

    val chatItems = arrayListOf(Item("Par défaut", 0, R.color.light_green))
    var chatItemSelected: Item = chatItems.first()

    var language: Language = Language.French
}

File: common/user-preferences.ts
export enum Language {
    French,
    English

}

export class Item{
    constructor(public name:string, public price: number, public theme: string,public description: string =''){}
}


export interface UserPreferences {

    appThemeSelected: string;
    boardItems: Item[];
    boardItemSelected: Item;
    chatItems: Item[];
    chatItemSelected: Item;
    language: Language;

}
